opendevin/sandbox/docker/ssh_box.py

```diff
@@ -137,6 +139,15 @@ def setup_user(self):
             )
             if exit_code != 0:
                 raise Exception(f'Failed to set password in sandbox: {logs}')
+
+            # chown the home directory
+            exit_code, logs = self.container.exec_run(
+                ['/bin/bash', '-c', 'chown opendevin:root /home/opendevin'],
+                workdir=SANDBOX_WORKSPACE_DIR,
+            )
+            if exit_code != 0:
+                raise Exception(
+                    f'Failed to chown home directory for opendevin in sandbox: {logs}')
```

Comment:
Maybe we want to pull this out in to the core Sandbox implementation, as `initialize_homedir`. Some sandboxes might override it, but most are going to do exactly this.

(TBH there's probably a lot of logic we could deduplicate between SSH, Exec, and Local)

Reply:
I agree! I was also thinking heavily about this while implementing this, maybe we should make a BaseClass to keep all the common logic (including this one!)

Reply:
well... just realized this might just work on `DockerSSHBox` and `ExecBox`, since in `LocalBox` we are just re-using the current user.

Reply:
Hmm good point. I could see putting this logic in the base class, and then having a `pass` override for the LocalBox that makes it a no-op

Reply:
I guess once we deprecate `ExecBox`, this logic will not be duplicated since `DockerSSHBox`, `LocalBox` and `E2BBox` will likely to be very different from each other in terms of implementation. So maybe we can leave this as is, and then try to deprecate `ExecBox` in the next few days/weeks?